### ISSUE TITLE
space-before-blocks: add option to specify the maximum amount of spaces before block

### ISSUE THREAD

**Issue by Thoro** [eslint/eslint#7023]
The space-before-blocks rule, should have an option to specify how many spaces before a block are acceptable, if there's already code on the same line (e.g. it shouldn't be affected by normal indentation).

**What version of ESLint are you using?**
v3.3.1

**What rule do you want to change?**
space-before-blocks

**What code should be flagged as incorrect with this change?**
```js
function test ()            {
}
```

**What happens when the rule is applied to this code now?**
No error message is printed.

**Comment by platinumazure** [comment]
Just for my curiosity, do you see anyone wanting to use any value other than 1 if they want to limit spaces?

I ask because that could result in an option `"disallowMultipleSpaces": true` (or similar) which might be more semantically useful in a config file than `"maxSpaces": 1`, but that comes with the tradeoff of rule flexibility.

**Comment by Thoro** [comment]
me personally, no

**Comment by platinumazure** [comment]
Forgot to mention, I'll champion this. (Assigning to myself so I can track this. Doesn't mean I'm necessarily going to work on it.)

@Thoro Would you be interested in submitting a PR if this issue gets accepted?

**Comment by Thoro** [comment]
Depends if I got time for this, in case I find resources on my side, I sure will make a PR.

**Comment by Thoro** [comment]
@platinumazure as mentioned in the PR the Commit Message gives me a headache, what should I make it like, so it's still a description and not just keywords? Or are keywords fine and it needs to be exactly 72 chars?

**Comment by mysticatea** [comment]
[no-multi-spaces](http://eslint.org/docs/rules/no-multi-spaces) rule has the responsibility to disallow multiple spaces.

**Comment by kaicataldo** [comment]
@mysticatea is right - I don't think this buys us anything that the `no-multi-spaces` rule doesn't already do.

**Comment by platinumazure** [comment]
I'd forgotten about that rule myself.

@Thoro Does `no-multi-spaces` cover your use case?

**Comment by Thoro** [comment]
@platinumazure No, doesn't look like it also handles spaces around blocks, at least my test cases throw no errors.

**Comment by mysticatea** [comment]
I think this should be the issue for `no-multi-spaces` rule rather than `space-before-blocks` rule.
Could you tell us your config (`.eslintrc`)?

**Comment by Thoro** [comment]
```
module.exports = {
    "rules": {
        "space-before-blocks": [ "error" ],
	"no-multi-space": [ "error" ]
    }
};
```

test file:

```
function World ()   {
}
```

**Comment by Thoro** [comment]
whoops, my have had an issue with my local build of eslint - sorry about that!

**Comment by ilyavolodin** [comment]
Closing. The issue has been resolved.
